src/preamble.lua

```diff
@@ -5,11 +5,34 @@ function Enum.__tostring(t)
 end
 
 function Enum.new(k, v)
-  local e = {k, v}
+  local e = { k, v }
   setmetatable(e, Enum)
   return e
 end
 
+RecordMeta = {}
+function RecordMeta.__tostring(record)
```

Comment:
I'd love to have tests for this, I can show you how to do that if you don't know how, but it's pretty straight forward. But you need a `tostring` foreign function and then just check the equality using pattern matches. Let me know if you need help. :D